kotekan/kotekan.cpp

```diff
@@ -463,8 +463,8 @@ int main(int argc, char** argv) {
 
     auto& metrics = prometheus::Metrics::instance();
     metrics.register_with_server(&rest_server);
-    auto& kotekan_running_metric = metrics.add_gauge("kotekan_running", "main");
-    kotekan_running_metric.set(running);
+    auto kotekan_running_metric = metrics.add_gauge("kotekan_running", "main", {});
+    kotekan_running_metric->labels({}).set(running);
```

Comment:
I've been trying to figure out what the point of this is (i.e. why call labels every time), and I think it's because you `.labels(...)` itself is leaky (i.e. it returns references to the `Metric` objects that are in theory owned by the `MetricsFamily` instance), and you're trying to avoid storing references that may be long lived.

This seems like it's also a fragile point (although probably much less fragile) and so if you're going to fix it you should do it properly. I think someone could still in theory bind a reference to a raw `Metric` object in an intialization list and that may outlive the `MetricsFamily` objects it belonged to, or the `MetricsFamily` could delete the instance from its `deque` (though no deletions are done at the moment, that's not to say they will never be done).

Potential resolutions to this I can imagine:
- labels should return a `std::shared_ptr<Metric>` and `MetricsFamily` should store shared pointers.
- labels should return an uncopyable proxy object that holds a `shared_ptr` to metrics family (this won't deal with the issue of deletion of the Metric from the MetricFamily).

Not sure if there's any other options. I wondered about trying to remove the move assignment operator to prevent binding a reference to the temporary returned by labels, but I don't think that works at all.